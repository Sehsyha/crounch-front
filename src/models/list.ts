import { Product } from './product';
import { CategoryInGetListResponse } from './category';

interface List {
  id: string
  name: string
  products: Product[]
}

interface GetListResponse {
  id: string
  name: string
  creationDate: string
  categories: CategoryInGetListResponse[]
}

export { List, GetListResponse }
