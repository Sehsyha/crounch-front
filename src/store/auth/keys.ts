export enum AuthActions {
  LOGIN = 'LOGIN',
  SIGNUP = 'SIGNUP',
  LOGOUT = 'LOGOUT',
}

export enum AuthGetters {
  IS_AUTHENTICATED = 'IS_AUTHENTICATED',

}

export enum AuthMutations {
  LOGIN = 'LOGIN',
  LOGOUT = 'LOGOUT',
}
