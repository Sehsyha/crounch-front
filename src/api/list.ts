import { doFetch } from './api';
import { List } from '@/models/list';

export function createList(name: string): Promise<List> {
  return doFetch({
    url: 'lists',
    method: 'POST',
    data: { name }
  });
}

export function getOwnerLists(): Promise<Array<List>> {
  return doFetch({
    url: 'lists',
    method: 'GET',
  });
}

export function deleteList(id: string): Promise<void> {
  return doFetch({
    url: `lists/${id}`,
    method: 'DELETE',
  });
}

export function addProductToList(productID: string, listID: string): Promise<void> {
  return doFetch({
    url: `lists/${listID}/products/${productID}`,
    method: 'POST'
  })
}

export function readList(listID: string): Promise<List> {
  return doFetch({
    url: `lists/${listID}`,
    method: 'GET'
  })
}
