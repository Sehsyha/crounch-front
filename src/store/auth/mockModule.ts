import { AuthKeys } from './keys';
import { Module } from 'vuex';
import { AuthState } from '.';
import { RootState } from '..';

const actions = {
  [AuthKeys.SIGNUP]: jest.fn(),
  [AuthKeys.LOGIN]: jest.fn(),
  [AuthKeys.LOGOUT]: jest.fn(),
};

const getters = {
  isAuthenticated: jest.fn(),
};

export function createAuthModuleMock(): Module<AuthState, RootState> {
  return {
    namespaced: true,
    actions,
    getters,
    state: {
      token: '',
      status: '',
    },
  };
}
