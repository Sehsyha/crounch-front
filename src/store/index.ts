import Vue from 'vue';
import Vuex, { StoreOptions } from 'vuex';
import { auth } from './auth';
import { ListModule } from './ListModule';
import { product } from './product'

Vue.use(Vuex);

export interface RootState { }

const store: StoreOptions<RootState> = {
  state: {},
  modules: {
    auth,
    ListModule,
    product,
  },
  strict: true,
};

export default new Vuex.Store<RootState>(store);
