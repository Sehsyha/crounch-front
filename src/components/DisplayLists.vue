<template>
  <div>
    <h2>Listes</h2>
    <ul>
      <li v-for="list in lists" :key="list.id" :id="list.id" class="list">
        <span @click="goToList(list.id)">{{ list.name }}</span>
        <span v-if="list.archivationDate"> Archiv√©e le {{ list.archivationDate }}</span>
        <button v-if="!list.archivationDate" @click="archiveList(list.id)">Archiver</button>
        <button @click="deleteList(list.id)">Supprimer</button>
      </li>
    </ul>
  </div>
</template>

<script lang="ts">
import { Vue, Component } from 'vue-property-decorator';
import { getModule } from 'vuex-module-decorators';

import { List } from '@/models/list';
import { ListModule } from '@/store/list/ListModule';

@Component
export default class DisplayLists extends Vue {
  public listModule: ListModule = getModule(ListModule)

  async created() {
     this.listModule.getUsers()
  }

  get lists () {
    return this.listModule.all
  }

  goToList(id: string) {
    this.$router.push(`/lists/${id}`)
  }

  async deleteList(id: string) {
   await this.listModule.deleteAction(id)
  }

  async archiveList(id: string) {
    await this.listModule.archiveList(id)
  }
};
</script>
