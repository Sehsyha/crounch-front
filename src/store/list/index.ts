import { Module } from 'vuex';

import { actions } from './actions';
import { RootState } from '..';
import { List } from '@/models/list';

export interface ListState {
  lists: List[];
}

export const state: ListState = {
  lists: [],
};

const namespaced: boolean = true;
export const listNamespace = { namespace: 'list' };

export const list: Module<ListState, RootState> = {
  namespaced,
  state,
  actions,
};
