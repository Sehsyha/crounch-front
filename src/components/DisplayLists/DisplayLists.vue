<template>
  <div>
    <h2>Listes</h2>
    <ul>
      <li
        v-for="list in this.lists"
        :key="list.id"
        :id="list.id"
        class="list"
        @click="goToList(list.id)"
      >{{ list.name }}</li>
    </ul>
  </div>
</template>

<script lang="ts">
import { Vue, Component } from 'vue-property-decorator';
import { Getter, Action } from 'vuex-class';

import { ListState, listNamespace } from '../../store/list';
import { List } from '../../models/list';
import { ListActions, ListGetters } from '../../store/list/keys';

@Component
export default class DisplayLists extends Vue {
  @Action(ListActions.GETOWNERS, listNamespace) getOwners!: () => List;
  @Getter(ListGetters.GETALL, listNamespace) lists!: List[];

  created() {
    this.getOwners();
  }

  goToList(id: string) {
    this.$router.push(`/lists/${id}`)
  }
};
</script>
