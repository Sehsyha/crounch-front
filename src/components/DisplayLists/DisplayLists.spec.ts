import { Module } from 'vuex';
import { Wrapper } from '@vue/test-utils';

import DisplayLists from './DisplayLists.vue';
import { shallowComponent } from '@/utils/test';
import { ListState } from '@/store/list';
import { RootState } from '@/store';
import { createListModuleMock } from '@/store/list/mockModule';
import { ListKeys } from '@/store/list/keys';
import { List } from '@/models/list';

describe('DisplayLists', () => {
  let wrapper: Wrapper<DisplayLists>;
  let list: Module<ListState, RootState>;

  beforeEach(() => {
    list = createListModuleMock();

    wrapper = shallowComponent(DisplayLists, {}, { list });
  });

  it('Should render.', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should get the list owners when the component is created.', () => {
    expect(list.actions![ListKeys.GETOWNERS]).toHaveBeenCalled();
  });

  it('Should display the returned lists.', () => {
    const lists: List[] = [
      {
        id: 'list id 1',
        name: 'list name 1',
      },
      {
        id: 'list id 2',
        name: 'list name 2',
      },
    ];

    (list.getters!.lists as jest.Mock).mockReturnValue(lists);

    wrapper = shallowComponent(DisplayLists, {}, { list });

    expect(wrapper).toMatchSnapshot();
  });
});
