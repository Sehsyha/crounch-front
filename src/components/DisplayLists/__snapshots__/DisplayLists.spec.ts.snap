// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayLists Should render. 1`] = `
<div>
  <h2>Listes</h2>
  <ul>
    <li id="list id 1" class="list">list name 1</li>
    <li id="list id 2" class="list">list name 2</li>
  </ul>
</div>
`;
